### ISSUE TITLE
Should ﷺ be identified as Arabic text?

### ISSUE THREAD

**Issue by zeerakahmed** [zeerakahmed/makhzan#10]
It's unclear whether the ﷺ ligature should be considered a regular part of Urdu text, or annotated as Arabic text.

**Comment by saadatm** [comment]
I would suggest that it should be considered Urdu text. If it’s annotated as Arabic, then its unligated form (صلّی اللّٰہ علیہ و سلم) should also be treated that way. But then there are other, similar phrases too, like علیہ السلام,‎ رضی اللّٰہ عنہ,‎ رحمۃ اللّٰہ علیہ,‎ ان شاء اللّٰہ,‎ ما شاء اللّٰہ,‎ and so on...  

Technically, all of the above are indeed Arabic, but their use in Urdu is widespread, and their inclusion in Urdu dictionaries (e.g. [صلّی اللّٰہ علیہ](https://archive.org/details/farhangiafiyah03amaduoft/page/n272/mode/2up) in _Farhang-e-Asifiyya_, and [ان شاء اللّٰہ](https://archive.org/details/IlmiUrduLughat-WarisSarhindi/page/n151/mode/2up) in _Ilmi Urdu Lughat_), is IMO a valid argument for keeping them as Urdu text.

However (and going a bit off-topic here), for text that _is_ annotated as Arabic (like the ayahs and hadith quoted in [0002.xml](https://github.com/zeerakahmed/makhzan/blob/master/text/0002.xml)):

There are certain characters that are common in Arabic and Urdu, but their Unicode codepoints are different. Compare ه (U+0647 ARABIC LETTER HEH) and ہ (U+06C1 ARABIC LETTER HEH GOAL); or ي (U+064A ARABIC LETTER YEH) and ی (U+06CC ARABIC LETTER FARSI YEH) (in these examples, Arabic uses the former character; Urdu, the latter). I haven’t inspected all of the Arabic text currently in Makhzan, but from what I have seen, such characters in Arabic text are using codepoints meant for Urdu. This is a common mistake made by Urdu authors when they are quoting Arabic, and I believe that it should be corrected in the corpus.

**Comment by zeerakahmed** [comment]
I agree with you. Most commonly used Arabic phrases, when used in Urdu text, we have identified as Urdu. There are instances when we see an obvious visual demarkation (such as a switch from Nastaliq to Naskh), that indicates that the author intends to identify the text as Arabic. In those cases we have relied on the visual demarcation. Where such demarcation does not exist, it makes sense to identify the text as Urud.

This particular ligature formed an interesting boundary, but I'm aligned that it is ok to identify this as Urdu text.

We have a large influx of text upcoming (we are in the process of cleaning it in another respository), and will aim to make this correction after it's added.

As for your second point, the use of Farsi/Urdu characters in Arabic text, we should make those fixes. I'll start a new issue where we track that.
